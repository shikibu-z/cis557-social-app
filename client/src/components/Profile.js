import React from 'react';
import {
  Row, Col,
} from 'react-bootstrap';
import '../stylesheets/Profile.css';

function Profile(props) {
  const {
    profileImage, userName, registrationDate, interests, gender, groupsInCommon,
  } = props;

  // list of groups, group[1] = groupname, group[0] = profile photo
  const interestsArray = [];
  for (let i = 0; i < interests.length; i += 1) {
    interestsArray.push(
      <p className="m-md-auto p-1 d-inline-block" key={interests[i]} style={{ 'text-decoration': 'underline' }}>
        {interests[i]}
      </p>,
    );
  }

  const groupsArray = [];
  for (let i = 0; i < groupsInCommon.length; i += 1) {
    groupsArray.push(
      <div key={i} style={{ display: 'inline-block' }}>
        <img alt="" id="profile-group-image" src={groupsInCommon[i][0]} title={groupsInCommon[i][1]} />
      </div>,
    );
  }

  return (
    <div className="profile">
      <img id="profile-avatar" src={profileImage} className="avatar" alt="profile" />
      <p id="profile-username">{userName}</p>
      <div id="profile-info">
        <Row>
          <Col md={2}>
            <div className="bulletpoint" />
          </Col>
          <Col md={10}>
            <p id="profile-gender"><strong>  Gender: </strong>   {gender}</p>
          </Col>
        </Row>
        <Row>
          <Col md={2}>
            <div className="bulletpoint" />
          </Col>
          <Col md={10}>
            <p id="profile-registration-date"><strong>  Registration Date: </strong>   {registrationDate}</p>
          </Col>
        </Row>
        <Row>
          <Col md={2}>
            <div className="bulletpoint" />
          </Col>
          <Col md={10}>
            <p id="profile-interest"><strong>  Interests: </strong> </p>
          </Col>
        </Row>
        <div className="d-flex flex-row justify-content-between">
          {interestsArray}
        </div>
        <Row>
          <Col md={2}>
            <div className="bulletpoint" />
          </Col>
          <Col md={10}>
            <p><strong>  Groups: </strong> </p>
          </Col>
        </Row>
        <div className="profile-array">
          {groupsArray}
        </div>
      </div>
    </div>
  );
}

export default Profile;
