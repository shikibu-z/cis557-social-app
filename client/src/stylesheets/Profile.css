.profile {
    position: absolute;
    top: 18%;
    width: 100%;
    height: 82%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.avatar {
    object-fit: cover;
    border-radius:50%;
    width: 150px;
    height: 150px;
}

#profile-username {

    font-family: Roboto;
    font-style: normal;
    font-weight: bold;
    font-size: 30px;
    line-height: 30px;

    color: rgba(31, 31, 31, 0.95);
}


#profile-group-image {
    object-fit: cover;
    border-radius:50%;
    width: 50px;
    height: 50px;
}

.bulletpoint {
    width: 25px;
    height: 25px;
    background: #F8B114;
    border-radius: 8px;
}

#profile-info {
    display: flex;
    flex-direction: column;
    gap: 5px;
}

.profile-row {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 5px;
}

.profile-array {
    display: flex;
    justify-content: center;
    flex-direction: row;
    gap: 10px;
}