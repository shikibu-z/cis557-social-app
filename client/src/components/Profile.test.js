/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import Profile from './Profile';

test('Profile component matches snapshot', () => {
  const component = renderer.create(<Profile
    profile_image="http://i.stack.imgur.com/Dj7eP.jpg"
    user_name="hello123"
    registrationDate="06/05/2021"
    interests={['sports', 'stock market']}
    gender="Male (He/Him/His)"
    groupsInCommon={[['https://cdn.vox-cdn.com/thumbor/ILugzkwUR0oRQFWskaSrcY3LJ4U=/0x0:4173x2582/1200x800/filters:focal(1754x958:2420x1624)/cdn.vox-cdn.com/uploads/chorus_image/image/66757988/1158933034.jpg.0.jpg', 'stock watcher'],
      ['https://static.nfl.com/static/content/public/static/img/navigation/shields/header-shield.png', 'NFL']]}
  />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
