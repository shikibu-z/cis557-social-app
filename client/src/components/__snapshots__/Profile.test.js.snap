// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component matches snapshot 1`] = `
<div
  className="profile"
>
  <img
    alt="profile"
    className="avatar"
    id="profile-avatar"
  />
  <p
    id="profile-username"
  />
  <div
    id="profile-info"
  >
    <div
      className="row"
    >
      <div
        className="col-md-2"
      >
        <div
          className="bulletpoint"
        />
      </div>
      <div
        className="col-md-10"
      >
        <p
          id="profile-gender"
        >
          <strong>
              Gender: 
          </strong>
             
          Male (He/Him/His)
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-2"
      >
        <div
          className="bulletpoint"
        />
      </div>
      <div
        className="col-md-10"
      >
        <p
          id="profile-registration-date"
        >
          <strong>
              Registration Date: 
          </strong>
             
          06/05/2021
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-2"
      >
        <div
          className="bulletpoint"
        />
      </div>
      <div
        className="col-md-10"
      >
        <p
          id="profile-interest"
        >
          <strong>
              Interests: 
          </strong>
           
        </p>
      </div>
    </div>
    <div
      className="d-flex flex-row justify-content-between"
    >
      <p
        className="m-md-auto p-1 d-inline-block"
        style={
          Object {
            "text-decoration": "underline",
          }
        }
      >
        sports
      </p>
      <p
        className="m-md-auto p-1 d-inline-block"
        style={
          Object {
            "text-decoration": "underline",
          }
        }
      >
        stock market
      </p>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-2"
      >
        <div
          className="bulletpoint"
        />
      </div>
      <div
        className="col-md-10"
      >
        <p>
          <strong>
              Groups: 
          </strong>
           
        </p>
      </div>
    </div>
    <div
      className="profile-array"
    >
      <div
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt=""
          id="profile-group-image"
          src="https://cdn.vox-cdn.com/thumbor/ILugzkwUR0oRQFWskaSrcY3LJ4U=/0x0:4173x2582/1200x800/filters:focal(1754x958:2420x1624)/cdn.vox-cdn.com/uploads/chorus_image/image/66757988/1158933034.jpg.0.jpg"
          title="stock watcher"
        />
      </div>
      <div
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <img
          alt=""
          id="profile-group-image"
          src="https://static.nfl.com/static/content/public/static/img/navigation/shields/header-shield.png"
          title="NFL"
        />
      </div>
    </div>
  </div>
</div>
`;
